File: engine/engine-model/src/main/scala/tools/packet/RecommendedWorldMessage.scala
package tools.packet

import net.opcodes.SendOpcode
import server.WorldRecommendation

class RecommendedWorldMessage(private var _worlds: java.util.List[WorldRecommendation]) extends PacketInput {
  def worlds: java.util.List[WorldRecommendation] = _worlds

  override def opcode(): SendOpcode = SendOpcode.RECOMMENDED_WORLD_MESSAGE
}

File: engine/engine-model/src/main/java/tools/packet/RecommendedWorldMessage.java
package tools.packet;

import java.util.List;

import net.opcodes.SendOpcode;
import server.WorldRecommendation;

public record RecommendedWorldMessage(List<WorldRecommendation> worlds) implements PacketInput {
  @Override
  public SendOpcode opcode() {
    return SendOpcode.RECOMMENDED_WORLD_MESSAGE;
  }
}